src/client/executor.rs

```diff
@@ -106,7 +110,25 @@ impl Client {
         op: &mut T,
         session: impl Into<Option<&mut ClientSession>>,
     ) -> Result<ExecutionDetails<T>> {
-        // Validate inputs that can be checked before server selection and connection checkout.
+        let session = session.into();
+        let ctx = self.start_operation_span(op, session.as_deref());
+        let result = self
+            .execute_operation_with_details_inner(op, session)
+            .with_context(ctx.clone())
+            .await;
+        #[cfg(feature = "opentelemetry")]
+        self.record_error(&ctx, &result);
```

Comment:
Will this record a duplicate error if an error is also recorded from the call to `record_command_result` on line 673?